engine/classes/Elgg/Notifications/NotificationsService.php

```diff
@@ -172,22 +187,25 @@ public function enqueueEvent($action, $type, $object) {
 			}
 
 			if ($registered) {
-				$this->queue->enqueue(new \Elgg\Notifications\Event($object, $action));
+				$this->queue->enqueue(new SubscriptionNotificationEvent($object, $action));
 			}
 		}
 	}
 
 	/**
 	 * Pull notification events from queue until stop time is reached
 	 *
-	 * @param int $stopTime The Unix time to stop sending notifications
-	 * @return int The number of notification events handled
+	 * @param int  $stopTime The Unix time to stop sending notifications
+	 * @param bool $matrix   If true, will return delivery matrix instead of a notifications event count
```

Comment:
Number of successful events tells us nothing. Matrix gives a better idea of what has been done.  I kept the old return to avoid rewriting tests.